src/app/ReviewService/ReviewService.cs

```diff
@@ -14,22 +15,22 @@ namespace ReviewService;
 public sealed class ReviewService<TReviewSettings> : IReviewService<TReviewSettings>
 	where TReviewSettings : ReviewSettings
 {
+	private readonly ILogger _logger;
 	private readonly IReviewPrompter _reviewPrompter;
 	private readonly IReviewSettingsSource<TReviewSettings> _reviewSettingsSource;
-	private readonly ReviewConditionCallback<TReviewSettings>[] _reviewConditions;
-	private readonly ILogger _logger;
+	private readonly IList<IReviewCondition<TReviewSettings>> _reviewConditions;
 
 	public ReviewService(
 		ILogger<ReviewService<TReviewSettings>> logger,
 		IReviewPrompter reviewPrompter,
 		IReviewSettingsSource<TReviewSettings> reviewSettingsSource,
-		ReviewConditionCallback<TReviewSettings>[] reviewConditions
+		IReviewConditionBuilder<TReviewSettings> reviewConditionStrategyBuilder
 	)
 	{
 		_logger = logger;
 		_reviewPrompter = reviewPrompter;
 		_reviewSettingsSource = reviewSettingsSource;
-		_reviewConditions = reviewConditions;
+		_reviewConditions = reviewConditionStrategyBuilder?.Conditions ?? ReviewConditionBuilder.Default<TReviewSettings>().Conditions;
```

Comment:
I think the solution would be to add a member to the builder interface. Basically a method to get the conditions list and that method would look at the current list and, if empty, return the default list.

Reply:
So that could be a `Build()` method on our builder 😋

Reply:
Nullable is not enable but intelisens what telling me that it can still be null and to do checks. Tbh I would not mess with nullable it adds a lot of noise into the solution.

Reply:
Maybe not now, but it would be good value. You should add a suggestion to the backlog.

Reply:
https://dev.azure.com/nventive/Practice%20committees/_workitems/edit/287200